OMCompiler/Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -2830,7 +2830,7 @@ algorithm
 
     case(DAE.CREF(cr, ty)::rest) equation
       slst = List.map(dims, intString);
-      var = SimCodeVar.SIMVAR(cr, BackendDAE.VARIABLE(), "", "", "", 0, NONE(), NONE(), NONE(), NONE(), false, ty, false, SOME(name), SimCodeVar.NOALIAS(), DAE.emptyElementSource, SimCodeVar.NONECAUS(), NONE(), slst, false, true, true, NONE(), NONE());
+      var = SimCodeVar.SIMVAR(cr, BackendDAE.VARIABLE(), "", "", "", 0, NONE(), NONE(), NONE(), NONE(), false, ty, false, SOME(name), SimCodeVar.NOALIAS(), DAE.emptyElementSource, SOME(SimCodeVar.NONECAUS()), NONE(), slst, false, true, true, NONE(), NONE(), NONE(), NONE());
```

Comment:
Why is the causality `NONECAUS()`?

Reply:
@lochel  To support some dummy variables or temporary variables

Reply:
Not sure what that means. There is no `NONECAUS()` in FMI 2.0. So are these FMI 1.0 specific variables?

Reply:
yes NONECAUS() are specific ti FMI 1.0

Reply:
@arun3688 I asked _why_ theses variables get declared with `causality=SimCodeVar.NONECAUS`. The function generates temporary variables for arrays which doesn't seem to be fmi 1.0 specific.

Reply:
i dont know, and it is not done for FMI 1.0, the fuction creates a temporary variable , with some default values and may be they update later, and we did not change anything new in that function just made the NONECAUS as Option